<launch>

    <node pkg="advanced_robotics_team6" type="fake_imu.py" name="imu_faker_node" />

    <node pkg="advanced_robotics_team6" type="imu_tf.py" name="imu_tf_node" />

    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_se" clear_params="true">
      <rosparam command="load" file="$(find advanced_robotics_team6)/params/ekf_localization.yaml"/>
    </node>

    <node name="imu_pid" pkg="pid" type="controller" output="screen" >
        <param name="Kp" value="-1.0" />
        <param name="Ki" value="-0.0" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="1905" />
        <param name="lower_limit" value="-1905" />
        <param name="windup_limit" value="10" />
        <param name="cutoff_frequency" value="12" />
        <param name="topic_from_controller" value="odroid/IMU/pid/control_effort" />
        <param name="topic_from_plant" value="odroid/IMU/pid/state" />
        <param name="setpoint_topic" value="odroid/IMU/pid/setpoint" />
        <param name="pid_enable_topic" value="odroid/IMU/pid/enable" />
        <param name="node_name" value="imu_pid" />
        <param name="max_loop_frequency" value="55.0" />
        <param name="min_loop_frequency" value="45.0" />
     </node>

    <node name="ir_bottom_pid" pkg="pid" type="controller" output="screen" >
        <param name="Kp" value="-7.0" />
        <param name="Ki" value="-3.0" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="1905" />
        <param name="lower_limit" value="-1905" />
        <param name="windup_limit" value="10" />
        <param name="cutoff_frequency" value="12" />
        <param name="topic_from_controller" value="odroid/bottom_IR/pid/control_effort" />
        <param name="topic_from_plant" value="odroid/bottom_IR/pid/state" />
        <param name="setpoint_topic" value="odroid/bottom_IR/pid/setpoint" />
        <param name="pid_enable_topic" value="odroid/bottom_IR/pid/enable" />
        <param name="node_name" value="ir_bottom_pid" />
        <param name="max_loop_frequency" value="55.0" />
        <param name="min_loop_frequency" value="45.0" />
     </node>

    <node name="ir_top_pid" pkg="pid" type="controller" output="screen" >
        <param name="Kp" value="-7.0" />
        <param name="Ki" value="-3.0" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="1905" />
        <param name="lower_limit" value="-1905" />
        <param name="windup_limit" value="10" />
        <param name="cutoff_frequency" value="12" />
        <param name="topic_from_controller" value="odroid/top_IR/pid/control_effort" />
        <param name="topic_from_plant" value="odroid/top_IR/pid/state" />
        <param name="setpoint_topic" value="odroid/top_IR/pid/setpoint" />
        <param name="pid_enable_topic" value="odroid/top_IR/pid/enable" />
        <param name="node_name" value="ir_top_pid" />
        <param name="max_loop_frequency" value="55.0" />
        <param name="min_loop_frequency" value="45.0" />
     </node>

     <node pkg="advanced_robotics_team6" type="wall_follow.py" name="odroid_node" output="screen" />


</launch>
